### DISCUSSION TITLE
MET python on derecho

### DISCUSSION THREAD

**Discussion by CPKalb** [dtcenter/METplus#2503] [Existing Builds] [answered]
Hi:

I am having an issue with python embedding on derecho.  When trying to use python embedding in Series-Analysis, I get the following error:

Traceback (most recent call last):
  File "/glade/work/dtcrt/METplus/derecho/components/MET/installations/11.1.0/share/met/python/met/dataplane.py", line 4, in <module>
    import netCDF4 as nc
  File "/glade/work/dtcrt/METplus/derecho/miniconda/miniconda3/envs/metplus_v5.1_py3.10/lib/python3.10/site-packages/netCDF4/__init__.py", line 3, in <module>
    from ._netCDF4 import *
ImportError: /glade/work/dtcrt/METplus/derecho/miniconda/miniconda3/envs/metplus_v5.1_py3.10/lib/python3.10/site-packages/netCDF4/../../../libnetcdf.so.19: undefined symbol: H5Pset_fapl_ros3
WARNING:
WARNING: tmp_nc_dataplane() -> an error occurred importing module "met.dataplane"
WARNING:
WARNING:
WARNING: get_series_entry() -> No match found for PYTHON in file: /glade/derecho/scratch/kalb/Stratosphere/METplus/model_applications/s2s/UserScript_fcstGFS_obsERA_StratosphereBias/FCST/FCST_zonal_mean_U_T_20180201_000000.nc
WARNING:
ERROR  :
ERROR  : DataPlane::set_constant(double) -> no data buffer allocated!
ERROR  :

It looks like something is going wrong with the netCDF4, but I'm unclear exactly what the issue is.  This import and script was working fine on cheyenne, but is not running on derecho.  Any ideas on how I would fix this?

Thanks,
Tina

**Comment by jprestop** [comment]
Hi @CPKalb.  Can you please try setting MET_PYTHON_EXE to `/glade/work/dtcrt/METplus/derecho/miniconda/miniconda3/envs/metplus_v5.1_py3.10/bin/python3` and rerun?  (Maybe first check to make sure I have the correct location?)

Please let me know if that resolves your problem, and if it does, sometime tomorrow, I'll add it to the module file that you load for MET.

**Reply by jprestop** [reply]
Hi @CPKalb.  I just wanted to follow up and see if you were able to try setting MET_PYTHON_EXE and if that worked?

**Reply by CPKalb** [reply]
I haven't had the chance to try this yet, but I will likely do so tomorrow.  Sorry for the delay.

**Comment by CPKalb** [comment]
I set MET_PYTHON_EXE to the path listed above as an environment variable, but am still getting the same error as before:

Traceback (most recent call last):
  File "/glade/work/dtcrt/METplus/derecho/components/MET/installations/11.1.0/share/met/python/met/dataplane.py", line 4, in <module>
    import netCDF4 as nc
  File "/glade/work/dtcrt/METplus/derecho/miniconda/miniconda3/envs/metplus_v5.1_py3.10/lib/python3.10/site-packages/netCDF4/__init__.py", line 3, in <module>
    from ._netCDF4 import *
ImportError: /glade/work/dtcrt/METplus/derecho/miniconda/miniconda3/envs/metplus_v5.1_py3.10/lib/python3.10/site-packages/netCDF4/../../../libnetcdf.so.19: undefined symbol: H5Pset_fapl_ros3

Tina

**Reply by jprestop** [reply, marked as answer]
@CPKalb thanks for trying it.  I'll need to recompile MET.  Do you have a deadline for when you need it reinstalled?

**Reply by CPKalb** [reply]
Not at the moment.  It's for a use case, but I can work on the other case until it's recompiled.

Thanks,
Tina

**Comment by jprestop** [comment]
HI @CPKalb.  Thank you for your patience.  I have reinstalled MET on Derecho.  Please retest and let us know how it goes.

**Reply by CPKalb** [reply]
I retested and everything seems to be working as expected.

Thanks,
Tiuna

**Reply by jprestop** [reply]
Great! Thanks for letting us know and for choosing an answer for this discussion.
